libvast/builtins/operators/drop.cpp

```diff
@@ -129,6 +131,19 @@ class plugin final : public virtual pipeline_operator_plugin {
       return config.error();
     return std::make_unique<drop_operator>(std::move(*config));
   }
+
+  virtual std::pair<std::string_view::iterator,
+                    caf::expected<std::unique_ptr<pipeline_operator>>>
+  parse_pipeline_string(std::string_view str) const override {
+    record options;
+    auto parse_result = system::parse_pipeline(str);
+    if (parse_result.parse_error) {
+      return {parse_result.new_str_it, parse_result.parse_error};
+    }
+    options["fields"] = parse_result.extractors;
+    options["schemas"] = parse_result.extractors;
+    return {parse_result.new_str_it, make_pipeline_operator(options)};
+  }
```

Comment:
Let's take a step back to the drawing board for how we write these parsers such that they are maintainable. I think we should re-use the basics from parseable in this:

```cpp
// Turn the remaining pipeline into an iterator pair.
auto f = str.begin();
const auto l = str.end();

// Assemble our parser.
using parsers::space, parsers::alnum, parsers::chr, parsers::eoi;
const auto ws = ignore(*space);
const auto extractor_char = alnum | chr{'_'} | chr{'-'} | chr{':'};
const auto extractor = !(':'_p | '-') >> (+extractor_char % '.');
const auto extractor_list = (extractor % (',' >> ws));
const auto p = extractor_list >> ws >> (eoi | '|');

// Parse!
auto extractors = std::vector<std::string>{};
if (!p(f, l, extractors))
  return {f, caf::make_error(...)};
if (extractors.empty())
  return {f, caf::make_error(...)};
return {f, std::make_unique<drop_operator>(std::move(extractors))};
```

Do you want to pair on this?